xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/internal/DefaultEntityResolver.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.xml.internal;
+
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.xml.EntityResolver;
+
+/**
+ * Uses css4j's entity resolver.
+ *
+ * @version $Id$
+ */
+@Component
+@Singleton
+public class DefaultEntityResolver extends io.sf.carte.doc.xml.dtd.DefaultEntityResolver implements EntityResolver
+{
+    public DefaultEntityResolver() {
+        super(false);
+        registerSystemIdFilename("http://jackrabbit.apache.org/dtd/graffito-jcr-mapping.dtd",
```

Comment:
Could you explain why there's this here? It feels a bit strange that there's only 1 entry standalone here. I think it deserves some comments.

WDYT?

Reply:
A new version of the commit explains why. I looked for DTDs in all the XWiki repositories, finding the XHTML 1.0 `Transitional` and `Strict`, and the _graffito_ DTD. My resolver has all the XHTML 1.x DTDs built-in, so only the _graffito_ had to be dealt with.

It is worth mentioning that, as you did not include the XHTML 1.1 DTDs with XWiki, any attempt to use XHTML 1.1 would result in requests being sent to `w3.org` which is applying 15s delays to petitions:

```http
$ curl -I http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0  2611    0     0    0     0      0      0 --:--:--  0:00:15 --:--:--     0HTTP/1.1 200 OK
date: Thu, 01 Oct 2020 11:31:34 GMT
last-modified: Tue, 20 Mar 2018 05:54:52 GMT
cache-control: max-age=31536000
expires: Fri, 01 Oct 2021 11:31:34 GMT
vary: Accept-Encoding
strict-transport-security: max-age=15552000; includeSubdomains; preload
access-control-allow-origin: *
content-length: 2611
content-type: text/plain
etag: W/"**********************************"
accept-ranges: bytes
```
As a side-note, they aren't using a DTD `Content-Type`, instead they use `text/plain`. I have yet to see a DTD being served under a DTD type.

Reply:
> It is worth mentioning that, as you did not include the XHTML 1.1 DTDs with XWiki,

That's a good point. We use HTML5 in most generated HTML so we probably don't need it. But CKEditor still uses XHTML 1.1 I think so it's possible that we're having extra delays there because of it.

Reply:
> We use HTML5 in most generated HTML

If you process that HTML5 with a normal XML toolchain (I mean with an XML parser), you need a resolver like mine which implements `EntityResolver2`. The PDF export process (which already uses the resolver) can resolve the entities thanks to that.